PowerEditor/src/ScitillaComponent/AutoCompletion.cpp

```diff
@@ -46,15 +46,53 @@ static bool isAllDigits(const generic_string &str)
 	return std::all_of(str.begin(), str.end(), ::isdigit);

```

Comment:
```_tcstol``` for each character (because string may be large enough to convert into max possible integer) might an alternative. (using lamda)
```C++
TCHAR *dummy = nullptr;
TCHAR input[] = _T("5x");
long l = _tcstol(input, &dummy , 10);
if (dummy && _tcsclen(dummy ))
{
          cout<<"Not a valid string to convert to integer";
}
```